lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/UpdateTaskHost.cs

```diff
@@ -104,19 +107,45 @@ public byte[] Call(byte[] memento)
                 }
             }
 
-            MapInputWithControlMessage<TMapInput> stopMessage =
+            if (!_break)
+            {
+                MapInputWithControlMessage<TMapInput> stopMessage =
                     new MapInputWithControlMessage<TMapInput>(MapControlMessage.Stop);
-            _dataAndControlMessageSender.Send(stopMessage);
+                _dataAndControlMessageSender.Send(stopMessage);
+            }
 
             _resultHandler.Dispose();
             return null;
         }
 
         /// <summary>
+        /// Received close event. If it is closed command sent from driver, set _break to true to stop the iteration in Call method
+        /// </summary>
+        /// <param name="value"></param>
+        public void OnNext(ICloseEvent value)
+        {
+            if (value.Value != null && value.Value.Value != null && Encoding.UTF8.GetString(value.Value.Value).Equals(TaskManager.CloseTaskByDriver))
+            {
+                Logger.Log(Level.Info, "The task received close event with message:", value.Value.Value);
```

Comment:
Missing format `{0}`. Also need to invoke with `string`, otherwise it will just print the address of the byte array.